openproblems/tasks/spatial_decomposition/api.py

```diff
@@ -0,0 +1,73 @@
+from ...data.sample import load_sample_data
+
+import numpy as np
+import pandas as pd
+
+EPS = 1e-10
+
+
+def check_dataset(adata):
+    """Check that dataset output fits expected API."""
+    # test for spatial coordinates
+    assert "spatial" in adata.obsm
+    # check that proportions are included
+    assert "proportions_true" in adata.obsm
+    # make sure proportions sum to one, some precision error allowed
+    assert np.all(np.abs(np.sum(adata.obsm["proportions_true"], axis=1) - 1) < EPS)
```

Comment:
In my experience converting to `float64` helps fix normalization error (not summing up to 1). We might consider doing that. For now, this works too.

Reply:
I see your point, but would this also work if the provided dataset originally stored the values in float32 - my thought was that the loss of precision would already have occurred then and we wouldn't be able to retrieve more significant digits.

```python
np.random.seed(3)
arr1 = np.random.random(10).astype(np.float32)
arr2 = arr1 / arr1.sum()
arr3 = arr2.astype(np.float64)
print(arr3.sum())
```
gives me the value `0.9999999636784196` which implies that float-type conversion might not be sufficient in these cases.

Reply:
I see, yes, `np.all` is best in this case I guess.